cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/system/CDAPLogAppender.java

```diff
@@ -151,9 +153,22 @@ public void start() {
           Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("log-clean-up"));
         FileMetadataCleaner fileMetadataCleaner = new FileMetadataCleaner(context.getTransactionRunner());
         LogCleaner logCleaner = new LogCleaner(fileMetadataCleaner, context.getLocationFactory(),
+                                               logFileManager.getLogsDirectoryLocation(),
                                                TimeUnit.DAYS.toMillis(fileRetentionDurationDays),
+                                               FOLDER_CLEANUP_BATCH_SIZE,
                                                fileCleanupBatchSize);
-        scheduledExecutorService.scheduleAtFixedRate(logCleaner, 10, logCleanupIntervalMins, TimeUnit.MINUTES);
+        scheduledExecutorService.schedule(new Runnable() {
+          @Override
+          public void run() {
+            long delayInMillis = logCleaner.run();
+            if (delayInMillis < 0) {
+              scheduledExecutorService.schedule(this, logCleanupIntervalMins, TimeUnit.MINUTES);
+            } else {
+              scheduledExecutorService.schedule(this, delayInMillis, TimeUnit.MILLISECONDS);
+            }
+          }
+        }, 0, TimeUnit.SECONDS);
```

Comment:
You can call `execute` instead of `schedule` to save the extra parameters

Reply:
Done